contrib/vagrant/start.sh

```diff
@@ -576,36 +576,91 @@ function vboxnet_addr_finder(){
+    create_master
+    create_workers
+    set_vagrant_env
+    create_k8s_config
 
-PROVISION_ARGS=""
-if [ -n "${NO_PROVISION}" ]; then
-    PROVISION_ARGS="--no-provision"
-fi
-if [ -n "${RELOAD}" ]; then
-    vagrant reload $PROVISION_ARGS $1
-elif [ -n "${PROVISION}" ]; then
-    vagrant provision $1
-else
-    vagrant up $PROVISION_ARGS $1
-    if [ "$?" -eq "0" -a -n "${K8S}" ]; then
-        host_port=$(vagrant port --guest 6443 k8s1)
-        vagrant ssh k8s1 -- cat /home/vagrant/.kube/config | sed "s;server:.*:6443;server: https://k8s1:$host_port;g" > vagrant.kubeconfig
-        echo "Add '127.0.0.1 k8s1' to your /etc/hosts to use vagrant.kubeconfig file for kubectl"
+    cd "${dir}/../.."
+
+    PROVISION_ARGS=""
+    if [ -n "${NO_PROVISION}" ]; then
+        PROVISION_ARGS="--no-provision"
     fi
-fi
+    if [ -n "${RELOAD}" ]; then
+        vagrant reload $PROVISION_ARGS $1
+    elif [ -n "${PROVISION}" ]; then
+        vagrant provision $1
+    else
+        vagrant up $PROVISION_ARGS $1
+        if [ "$?" -eq "0" -a -n "${K8S}" ]; then
+            host_port=$(vagrant port --guest 6443 k8s1)
+            vagrant ssh k8s1 -- cat /home/vagrant/.kube/config | sed "s;server:.*:6443;server: https://k8s1:$host_port;g" > vagrant.kubeconfig
+            echo "Add '127.0.0.1 k8s1' to your /etc/hosts to use vagrant.kubeconfig file for kubectl"
+        fi
+    fi
+}
 
+#get number of running VM(s)
+runningVm=$(VBoxManage list runningvms | awk 'END{ print NR }')
+VMName=$(VBoxManage list runningvms | awk 'NR==1{print $1}' |  cut -d "\"" -f 2)
+if [ "$VMName" ]; then
+    echo
+	echo "Detected running VMs that might cause conflict"
+	echo
+    VBoxManage list runningvms
+
+	#option to stop, delete VM(s) or ignore and continue
+    echo
+	printf "Do you wish to stop, destroy the VM(s) or ignore and continue? [s/d/C]  "
+	read optn
+    
+	case "$optn" in
+		"s" )
+            #stop the VM(s)
+			for ((i=1; i<=$runningVm; i=i+1))
+			do
+				VMName=$(VBoxManage list runningvms | awk 'NR==1{print $1}' |  cut -d "\"" -f 2)
+				echo
+				VBoxManage controlvm $VMName poweroff
+				printf "\n$VMName stopped\n"
+			done
+			printf "\n$runningVm VM(s) stopped successfully\n"
+			printf "Tip: Try running vagrant status to check status\n\n"
+		;;
+		"d" )
+            #destroy the VM(s)
+			for ((i=1; i<=$runningVm; i=i+1))
+			do
+				VMName=$(VBoxManage list runningvms | awk 'NR==1{print $1}' |  cut -d "\"" -f 2)
+				echo
+				VBoxManage controlvm $VMName poweroff
+				VBoxManage unregistervm $VMName --delete				
+				printf "\n$VMName destroyed\n"				
+			done
+			printf "\n$runningVm VM(s) destroyed successfully\n"
+			printf "Tip: Try running vagrant status to check status\n\n"
+		;;
+		"C" | * )
+            #continue and default case
+            echo
+            #value to start.sh is passed to function createVm
+			createVm $1
+		;;
+	esac
+else
+    echo
+    #value to start.sh is passed to function createVm
+	createVm $1
+fi 
```

Comment:
Missing newline here.

Reply:
added the missing newline and fixed style issues